sprint_tasks/MSI_2/MSI_2_send_message_ms_teams_dag.py

```diff
@@ -0,0 +1,163 @@
+    MESSAGE_COLOR = "#F8C471"
+
+    def __init__(self, webhook_url):
+        self.webhook_url = webhook_url
+
+    def send_message(self, title, quote_text, quote_author, image_url):
+        """Method combines the whole message and send it to MSTeams channel."""
+        myTeamsMessage = pymsteams.connectorcard(self.webhook_url)
+        myTeamsMessage.color(self.MESSAGE_COLOR)
+        myTeamsMessage.title(title)
+        myTeamsMessage.text(
+            f"**Quote:** {quote_text}\n\n"
+            f"**Author:** {quote_author}\n\n"
+            f"![Image]({image_url})"
+        )
+
+        try:
+            myTeamsMessage.send()
+            logging.info("Message successfully sent.")
+            return
+        except (ConnectionError, HTTPError, TimeoutError) as e:
+            logging.error(
+                f"Failed to send message due to network error: {e}",
+                exc_info=True
+            )
+        except pymsteams.TeamsWebhookException as e:
+            logging.error(
+                f"Failed to send message due to Teams webhook error: {e}",
+                exc_info=True
+            )
+        except ValueError as e:
+            logging.error(
+                f"Failed to send message due to value error: {e}",
+                exc_info=True
+            )
+
+
+class MessageSenderExecutor:
+    def __init__(self, quote_generator, image_generator, message_sender):
+        self.quote_generator = quote_generator
+        self.image_generator = image_generator
+        self.message_sender = message_sender
+
+    def run_process(self):
+        quote_content = self.quote_generator.get_content()
+        if not quote_content:
+            logging.error("Failed to generate quote")
+            return None
+
+        quote_text, quote_author = quote_content
+
+        image_url = self.image_generator.get_content()
+        if not image_url:
+            logging.error("Failed to generate image.")
+
+        self.message_sender.send_message(
+            "Sent by Natalia Ananeva",
+            quote_text,
+            quote_author,
+            image_url
+        )
+
+
+def run_script():
+    # Create instances
+    quote_generator = QuoteGenerator()
+    image_generator = ImageGenerator()
+    message_sender = TeamsMessageSender(TEAMS_WEBHOOK_URL)
+
+    # Run process
+    executor = MessageSenderExecutor(
+        quote_generator,
+        image_generator,
+        message_sender
+    )
+    executor.run_process()
+
+
+with DAG(
+    dag_id='send_massage_msteams_dag',
+    start_date=datetime.datetime(
+        year=2024,
+        month=6, 
+        day=10
+    ),
+    schedule_interval="0 12 * * *",  # send every day at 12 a.m.
+    tags=['send_message_teams'],
+    description='A DAG to send message to MS Teams channel using webhook. \
+        Message consists of quote and picture. ',
+    catchup=False
+) as dag:
+
+    start_op = EmptyOperator(
+        task_id='start'
+    )
+
+    send_message_op = PythonOperator(
```

Comment:
Would it fail in case something goes wrong? (like a network error or site API not availiable)